src/lib/structures/UserPaginatedMessage.ts

```diff
@@ -0,0 +1,135 @@
+import { GuildMessage } from '#lib/types';
+import { MessageBuilder, MessagePage, PaginatedMessage, PaginatedMessageOptions } from '@sapphire/discord.js-utilities';
+import { Time } from '@sapphire/time-utilities';
+import { RESTPostAPIChannelMessageJSONBody } from 'discord-api-types/v6';
+import { APIMessage, MessageEmbed, MessageEmbedOptions, MessageOptions, NewsChannel, TextChannel, User } from 'discord.js';
+
+export class UserPaginatedMessage extends PaginatedMessage {
+	public template: MessageOptions;
+
+	public constructor(options: UserPaginatedMessage.Options = {}) {
+		super(options);
+		this.setIdle(Time.Minute * 5);
+		this.template = UserPaginatedMessage.resolveTemplate(options.template);
+	}
+
+	public async start(message: GuildMessage, target = message.author): Promise<this> {
+		// Stop the previous display and cache the new one
+		const display = UserPaginatedMessage.handlers.get(target.id);
+		if (display) display.collector!.stop();
+
+		// If the message was sent by Skyra, set the response as this one
+		if (message.author.bot) this.response = message;
+
+		const handler = await this.run(target, message.channel);
+		const messageID = handler.response!.id;
+
+		if (this.collector) {
+			this.collector!.once('end', () => {
+				UserPaginatedMessage.messages.delete(messageID);
+				UserPaginatedMessage.handlers.delete(target.id);
+			});
+
+			UserPaginatedMessage.messages.set(messageID, handler);
+			UserPaginatedMessage.handlers.set(target.id, handler);
+		}
+
+		return handler;
+	}
+
+	/**
+	 * This clones the current handler into a new instance.
+	 */
+	public clone(): UserPaginatedMessage {
+		const clone = super.clone() as UserPaginatedMessage;
+		clone.template = this.template;
+		return clone;
+	}
+
+	public addPageBuilder(cb: (builder: MessageBuilder) => MessageBuilder): this {
+		return this.addPage(cb(new MessageBuilder()));
+	}
+
+	public addPageEmbed(cb: (builder: MessageEmbed) => MessageEmbed): this {
+		return this.addPage({ embed: cb(new MessageEmbed()) });
+	}
+
+	/**
+	 * Sets up the message's reactions and the collector.
+	 * @param channel The channel the handler is running at.
+	 * @param author The author the handler is for.
+	 */
+	protected async setUpReactions(channel: TextChannel | NewsChannel, author: User): Promise<void> {
+		if (this.pages.length > 1) await super.setUpReactions(channel, author);
+	}
```

Comment:
Can't this be merged in upstream?

Reply:
wdym?

Reply:
aka move the if check in the main module

Reply:
If that behaviour is desirable, I can move it upstream.

However... what if somebody defines custom actions which do page-independent actions? Say an info page.

Reply:
Well that'd be a different page so.. 🤔

Reply:
Well... nothing stops them from extending the behaviour to add virtual pages which aren't on the `this.pages` array.

Reply:
Riight, but... what's the point of reacting with one page? imo, if you want that behavior, you WANT to extend anyways